pkg/registry/networking/ingress/storage/storage_test.go

```diff
@@ -28,13 +28,12 @@ import (
 	genericregistrytest "k8s.io/apiserver/pkg/registry/generic/testing"
 	etcdtesting "k8s.io/apiserver/pkg/storage/etcd/testing"
 	api "k8s.io/kubernetes/pkg/apis/core"
-	"k8s.io/kubernetes/pkg/apis/extensions"
 	"k8s.io/kubernetes/pkg/apis/networking"
 	"k8s.io/kubernetes/pkg/registry/registrytest"
 )
 
 func newStorage(t *testing.T) (*REST, *StatusREST, *etcdtesting.EtcdTestServer) {
-	etcdStorage, server := registrytest.NewEtcdStorage(t, extensions.GroupName)
+	etcdStorage, server := registrytest.NewEtcdStorage(t, networking.GroupName)
```

Comment:
because ingresses don't encode to the default version of the networking API group yet, we need to use the resource-specific storage lookup:
```
registrytest.NewEtcdStorageForResource(t, networking.Resource("ingresses"))
```

Reply:
Ahh, I thought I tried this out but looks like I hacked up a weird variation. I will try this out now. Thanks for all the help!

Reply:
looks like there's still a NPE error building the storage. this may be the first unit test to use a cohabitating resource stored in a non-default version. `NewEtcdStorageForResource()` is missing this line:
```
completedConfig.ApiResourceConfig = master.DefaultAPIResourceConfigSource()
```
but adding that creates an import cycle

looking into what we can do here

Reply:
add this to NewEtcdStorageForResource:

```diff
diff --git a/pkg/registry/registrytest/etcd.go b/pkg/registry/registrytest/etcd.go
index 8510010584..23c7aef7e0 100644
--- a/pkg/registry/registrytest/etcd.go
+++ b/pkg/registry/registrytest/etcd.go
@@ -25,6 +25,7 @@ import (
        "k8s.io/apiserver/pkg/storage/storagebackend"
        "k8s.io/kubernetes/pkg/api/testapi"
        "k8s.io/kubernetes/pkg/kubeapiserver"
+       serverstorage "k8s.io/apiserver/pkg/server/storage"
 )

 func NewEtcdStorage(t *testing.T, group string) (*storagebackend.Config, *etcdtesting.EtcdTestServer) {
@@ -43,6 +44,7 @@ func NewEtcdStorageForResource(t *testing.T, resource schema.GroupResource) (*st
        if err != nil {
                t.Fatal(err)
        }
+       completedConfig.ApiResourceConfig = serverstorage.NewResourceConfig()
        factory, err := completedConfig.New()
        if err != nil {
                t.Fatal(err)
```

Reply:
bullseye, this works like a charm now